### ISSUE TITLE
Ensure the fifo is removed on quit

### ISSUE THREAD

**Issue by mawww** [danr/kakoune-languageclient#9]
Whenever the plugin quits (gracefully or not), it should remove the temporary directory and the fifo. At the moment, when that happens, Kakoune can end up frozen as it tries to write to the fifo and the fifo is blocking as its buffer is full (and nobody reads it).

**Comment by danr** [comment]
I agree. However, you should be able to unfreeze it with Ctrl-C, right? (Even if the timer text has not appeared)

**Comment by mawww** [comment]
Yeah, Ctrl-C works in Kakoune, still having it freeze each time a `.` gets inserted is a bit inconvenient.
